go/cmd/dolt/commands/merge.go

```diff
@@ -369,6 +301,100 @@ func constructInterpolatedDoltMergeQuery(apr *argparser.ArgParseResults) (string
 	return interpolatedQuery, nil
 }
 
+// calculateMergeConflicts calculates the count of conflicts that occurred during the merge.
+func calculateMergeConflicts(queryist cli.Queryist, sqlCtx *sql.Context, mergeStats map[string]*merge.MergeStats) (map[string]*merge.MergeStats, bool, int) {
+	dataConflicts, err := GetRowsForSql(queryist, sqlCtx, "SELECT * FROM dolt_conflicts")
+	if err != nil {
+		cli.Println(err.Error())
+		return nil, false, 1
+	}
+	for _, conflict := range dataConflicts {
+		tableName := conflict[0].(string)
+		if ok := mergeStats[tableName]; ok != nil {
+			mergeStats[tableName].DataConflicts = int(conflict[1].(uint64))
+		} else {
+			mergeStats[tableName] = &merge.MergeStats{DataConflicts: int(conflict[1].(uint64))}
+		}
+	}
+
+	schemaConflicts, err := GetRowsForSql(queryist, sqlCtx, "SELECT * FROM dolt_schema_conflicts")
+	if err != nil {
+		cli.Println(err.Error())
+		return nil, false, 1
+	}
+	for _, conflict := range schemaConflicts {
+		tableName := conflict[0].(string)
+		if ok := mergeStats[tableName]; ok != nil {
+			mergeStats[tableName].SchemaConflicts = 1
+		} else {
+			mergeStats[tableName] = &merge.MergeStats{SchemaConflicts: 1}
+		}
+	}
+
+	constraintViolations, err := GetRowsForSql(queryist, sqlCtx, "SELECT * FROM dolt_constraint_violations")
+	if err != nil {
+		cli.Println(err.Error())
+		return nil, false, 1
+	}
+	for _, conflict := range constraintViolations {
+		tableName := conflict[0].(string)
+		if ok := mergeStats[tableName]; ok != nil {
+			mergeStats[tableName].ConstraintViolations = int(conflict[1].(uint64))
+		} else {
+			mergeStats[tableName] = &merge.MergeStats{ConstraintViolations: int(conflict[1].(uint64))}
+		}
+	}
+
+	return mergeStats, dataConflicts == nil && schemaConflicts == nil && constraintViolations == nil, 0
+}
+
+// calculateMergeStats calculates the table operations and row operations that occurred during the merge.
+func calculateMergeStats(queryist cli.Queryist, sqlCtx *sql.Context, mergeStats map[string]*merge.MergeStats) (map[string]*merge.MergeStats, int) {
+	diffSummaries, err := getDiffSummariesBetweenRefs(queryist, sqlCtx, "head^1", "head")
```

Comment:
Does this only get one side of the merge? I would think we want to get the merge stats of HEAD^1->HEAD and union them with the merge stats of HEAD^2 -> HEAD.

I don't want to rabbit hole on this, so fine to ship. But I'm curious what the current implementation does... and it definitely doesn't jump out at me when I read origin/main.

Reply:
I think we only care about the `HEAD^1` -> `HEAD` diff because we want to know what changed in the branch we're currently in. `HEAD^2` -> `HEAD` diff would tell us how `HEAD` differs from `HEAD` in the branch we merged in.